java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java

```diff
@@ -864,8 +864,8 @@ else if (value instanceof double[])
       return NetworkTablesJNI.putDoubleArray(pathWithSep + key, (double[])value);
     else if (value instanceof Boolean[])
       return NetworkTablesJNI.putBooleanArray(pathWithSep + key, toNative((Boolean[])value));
-    else if (value instanceof Double[])
-      return NetworkTablesJNI.putDoubleArray(pathWithSep + key, toNative((Double[])value));
+    else if (value instanceof Number[])
```

Comment:
All arrays subclass `Object`, so there's no boxing.

Take this simple method:

```java
public static boolean isNumberArray(Object arr) {
  return arr instanceof Number[];
}
```

This will return `true` if passed an array of any boxed number type, and `false` for everything else